plugins/GSdx/GSDevice.cpp

```diff
@@ -45,7 +45,7 @@ GSDevice::GSDevice()
 
 GSDevice::~GSDevice()
 {
-	for(auto &t : m_pool) delete t;
+	for(auto t : m_pool) delete t;
```

Comment:
Ok. Yes people (i.e. I) often forgot to add the `&` (which ought to be the default). But I agree it is the same here.

For completeness, could you extend fast list to support access by reference (if it is doable).